### PR TITLE
[REF+test] clean up code for getting labels for merge screen, stdise

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Primarily a code clean up & test add this has the side effect of rendering labels for preferred language & the greeting IDs - which seems to me like a positive side effect

Before
----------------------------------------
before

After
----------------------------------------
after

Technical Details
----------------------------------------
The whole getSpecialValues function was a convoluted way of
1) getting labels for 2 fields - preferred communication method & communication style
2) serializing the value for preferred communication method

It was deeply confusing & hence had to be cleaned up

Comments
----------------------------------------

@JKingsnorth

### CODE REVIEW

**Comment by JKingsnorth** [comment]
I will try to QA this, but it looks good from the screenshots. The 'addressee ID' thing is a bit weird; but 'better' than just displaying a meaningless ID I suppose.

File: CRM/Dedupe/Merger.php [hunk 1]
```diff
@@ -1092,14 +1090,22 @@ public static function getRowsElementsAndInfo($mainId, $otherId, $checkPermissio
         // CRM-15681 don't display sub-types in UI
         continue;
       }
-      foreach (['main', 'other'] as $moniker) {
-        $contact = &$$moniker;
```
Comment (JKingsnorth) [inline CRM/Dedupe/Merger.php]: Hooray! It's gone!

**Comment by JKingsnorth** [comment]
Yes, the changes look good to me. I tested the display, and the merging, and it worked as expected.

However - I did come across a new bug while testing this:
https://lab.civicrm.org/dev/core/issues/970

But it is not a regression, so I recommend merging this PR.

**Review by JKingsnorth — APPROVED** [review]

**Comment by colemanw** [comment]
@eileenmcnaughton this needs rebasing.

**Comment by eileenmcnaughton** [comment]
@colemanw done

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by seamuslee001** [comment]
I'm going to merge based on @JKingsnorth 's review but @eileenmcnaughton i assume you will want to follow up on that other issue @JKingsnorth has reported

**Comment by eileenmcnaughton** [comment]
@seamuslee001 dying to :-)

But yeah it's on my radar

**Pull request merged**